vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java

```diff
@@ -1118,4 +1174,336 @@ public AdditionalComponent(Class<? extends Buildable> p, Class<? extends Buildab
       return child;
     }
   }
+
+  /**
+   * Container for search parameters
+   */
+  private static class SearchParameters {
+    public static final String SEARCH_STRING = "searchString";
+    public static final String MATCH_CASE    = "matchCase";
+    public static final String MATCH_NAMES   = "matchNames";
+    public static final String MATCH_TYPES   = "matchTypes";        
+
+    /** Current search string */
+    private String searchString;
+
+    /** True if case-sensitive */
+    private boolean matchCase;
+
+    /** True if match configurable names */
+    private boolean matchNames;
+
+    /** True if match class names */
+    private boolean matchTypes;
+    
+    /** Attach to our module preferences, if relevant */
+    private static Prefs prefs;    
+    
+    /**
+     * Constructs a new search parameters object, using the preferences.
+     */
+    public SearchParameters () {
+      // Attach to our module preferences if constructed this way. This also marks that we will write them when modified 
+      prefs = GameModule.getGameModule().getPrefs(); 
+      
+      prefs.addOption(null, new StringConfigurer(SearchParameters.SEARCH_STRING, null, ""));
+      prefs.addOption(null, new BooleanConfigurer(SearchParameters.MATCH_CASE,   null, false));
+      prefs.addOption(null, new BooleanConfigurer(SearchParameters.MATCH_NAMES,  null, true));
+      prefs.addOption(null, new BooleanConfigurer(SearchParameters.MATCH_TYPES,  null, true));
+      
+      this.searchString = (String) prefs.getValue(SearchParameters.SEARCH_STRING);
+      this.matchCase    = (Boolean)prefs.getValue(SearchParameters.MATCH_CASE);
+      this.matchNames   = (Boolean)prefs.getValue(SearchParameters.MATCH_NAMES);
+      this.matchTypes   = (Boolean)prefs.getValue(SearchParameters.MATCH_TYPES);                   
```

Comment:
You don't need any of the `this`es here.

Reply:
done